src/operators/ConvolutionOperator.cpp

```diff
@@ -55,17 +56,15 @@ void ConvolutionOperator<D>::initializeOperator(GreensKernel &greens_kernel) {
         println(10, "Time transform      " << trans_t);
         println(10, std::endl);
 
-        this->kern_exp.push_back(k_tree);
+        this->kern_exp.push_back(std::make_tuple(1.0, k_tree));
```

Comment:
I'm not crazy about this syntax. Is it important to make it clear that the (coef, tree) forms a tuple under the hood?

Reply:
Aha, this is by necessity, not by choice?

Reply:
`push_back` for `std::vector<T>` will only accept stuff of type `T` as arguments. The alternative is to have a free function: 
```
template <int D>
void push_back(FunctionVectorTree<D> &fs, const FuntionTree<D> * f, double coef = 1.0);
```
but then it looks a bit weird (I think) at call site: 
```
push_back(tree_vec, k_tree);
```
I have been looking at overloading `push_back` from `std::vector<T>` and it seems to be explicitly forbidden by the standard...

Reply:
Hmm, `CoefsFunctionTree<D>` has a constructor that takes `(double, FunctionTree<D> *)`. It's weird that this constructor is not picked up automatically if you call `push_back` with these two arguments (I've tried). Probably because of the template...